micrometer-support/src/main/java/io/javaoperatorsdk/operator/monitoring/micrometer/MicrometerMetrics.java

```diff
@@ -381,11 +373,79 @@ public MicrometerMetricsBuilder notCollectingMetricsPerResource() {
     }
 
     public MicrometerMetrics build() {
-      return new MicrometerMetrics(registry, Cleaner.NOOP, collectingPerResourceMetrics);
+      return new MicrometerMetrics(registry, cleanerBuilder.build(), collectingPerResourceMetrics);
+    }
+  }
+
+  public static class CleanerBuilder {
+
+    private final MeterRegistry registry;
+    private CleanerType cleanerType = CleanerType.NOOP;
+    private int cleaningThreadsNumber = 0;
+    private int cleanUpDelayInSeconds = 0;
+
+    public CleanerBuilder(MeterRegistry registry) {
+      this.registry = registry;
+    }
+
+    /**
+     * @param cleanerType the type of cleaner to use, defaults to {@link CleanerType#NOOP} if not
+     *     specified
+     */
+    public CleanerBuilder withCleanerType(CleanerType cleanerType) {
```

Comment:
I would rather not expose the type if possible but rather switch the type internally based on whether the user sets the number of cleaning threads or the delay, which would switch the implementation rather than throwing an exception if the type is not amenable to set these values.

Reply:
That might actually work, yes! With that we might not need the types at all. The `build()` method could just check if a delay parameter is set, otherwise use Cleaner.NOOP. The `DefaultCleaner` is actually not exposed right now at all, so no need to change that.

Reply:
I've updated the PR to get rid of the CleanerType and derive the type dynamically from the set builder parameters.